acm/odh-edge/apps/telemetry/otel-collector.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: cluster-collector
+  namespace: ai-edge-collector
+  labels:
+    app.kubernetes.io/managed-by: opentelemetry-operator
+spec:
+  mode: deployment
+  observability:
+    metrics:
+      enableMetrics: true
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+            - job_name: 'otel-collector'
+              bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+              scrape_interval: 5s
+              metrics_path: "/metrics"
+              static_configs:
+                - targets: ['tensorflow-housing-app-model-1.tensorflow-housing-app.svc.cluster.local:8080', 'bike-rental-app-model-1.bike-rental-app.svc.cluster.local:8082']
```

Comment:
This is currently very specific and static to the example applications that are included in this repo. Do you have any ideas in mind for how elements in this list would get added/updated/deleted, or am I thinking about it the wrong way?

My very vague/uninformed understanding was that OpenTelemetry worked more in a push rather than pull manner, where applications would connect to an OTel endpoint and send their telemetry data to that, rather than being scraped by the centralized system. Does that sound correct? If so, is that the direction we're planning to go, rather than the above static scrape configs?

Reply:
> where applications would connect to an OTel endpoint and send their telemetry data to that

That can be correct if the app in question is instrumented to support otel. In our case we just have prometheus endpoints so we use the prometheus receiver in otel which scrapes the endpoint. 

> This is currently very specific and static to the example applications

Yes, I agree. A better solution is to use labels to define what should be scraped but I had a bunch of issues configuring that. This PR at least gets the otel collector in place and metrics flowing and I can revisit to define a config that will scrape based on labels which is a preferred solution. 

I could document that a user would need to edit that static list for now?